astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -178,6 +178,13 @@ def function_to_method(n, klass):
     return n
 
 
+def attach_to_parent(node: NodeNG, name: str, parent: NodeNG):
+    frame = parent.frame()
+    frame.set_local(name, node)
+    if frame is parent:
+        frame._append_node(node)
```

Comment:
This feels like a crucial change that we don't have a test for. Should we add one?

Reply:
So, usually the classes get added to the body of the parent in the positinit constructor. There is only one exception to my knowledge: `raw_building._add_dunder_class`. It was added solely to support some `pyqt` inference and it is creating (and attaching) adhoc classes for the purposes of being later detected by the transform process. 

That situation is already tested by the three qt tests (which fail without this change). In other situations it's useless, afaics.

Overall, the process described above seems very perverse, the "dunder" class shouldn't be recreated within a function each time. It is its own class that is defined outside. We should rather consult the type annotation in the transform process. However, it is a question for another PR.